packages/contracts/contracts/voting/token/TokenVoting.sol

```diff
@@ -2,23 +2,23 @@
 
 pragma solidity 0.8.10;
 
-import {ERC20VotesUpgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import {IVotesUpgradeable} from "@openzeppelin/contracts-upgradeable/governance/utils/IVotesUpgradeable.sol";
 
 import {MajorityVotingBase} from "../majority/MajorityVotingBase.sol";
 import {IDAO} from "../../core/IDAO.sol";
 import {IMajorityVoting} from "../majority/IMajorityVoting.sol";
 
-/// @title ERC20Voting
+/// @title TokenVoting
 /// @author Aragon Association - 2021-2022
-/// @notice The majority voting implementation using an ERC-20 token.
+/// @notice The majority voting implementation using an [OpenZepplin `Votes`](https://docs.openzeppelin.com/contracts/4.x/api/governance#Votes) compatible governance token.
 /// @dev This contract inherits from `MajorityVotingBase` and implements the `IMajorityVoting` interface.
-contract ERC20Voting is MajorityVotingBase {
+contract TokenVoting is MajorityVotingBase {
     /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
-    bytes4 internal constant ERC20_VOTING_INTERFACE_ID =
+    bytes4 internal constant TOKEN_VOTING_INTERFACE_ID =
```

Comment:
It would be good that supportsInterface can give us the answer whether it's erc20 or nft based.
nno idea how yet..

Reply:
`TokenVoting` will work the same way regardless if it uses and ERC20 or NFT.

Maybe we should make `IVotesUpgradeable private votingToken;` public so that someone can check it from the outside?